azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,958 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/monitor/mgmt/2017-05-01-preview/insights"
```

Comment:
*Suggestion* @tombuildsstuff @katbyte This package seems to be [deprecated by Azure Go SDK team](https://github.com/Azure/azure-sdk-for-go/blob/master/services/monitor/mgmt/2017-05-01-preview/insights/autoscalesettings.go#L28). Can we use the [new one](https://github.com/Azure/azure-sdk-for-go/blob/master/services/preview/monitor/mgmt/2018-03-01/insights/autoscalesettings.go) instead?